onnxruntime/core/graph/graph_utils.cc

```diff
@@ -445,6 +465,65 @@ size_t RemoveNodeOutputEdges(Graph& graph, Node& node) {
   return output_edges.size();
 }
 
-}  // namespace graph_utils
+void ReplaceNodeOutput(Graph& graph, Node& node, Node& replacement, int replacement_output_idx) {
```

Comment:
This method actually connects the incoming node of a node with its outgoing nodes, so I think it needs a different name. Also, I think we should not call the RemoveGraphEdges in here, it does not fit conceptually. Connecting the incoming with the outgoing nodes of a node should not affect the node itself.

Reply:
I believe they need to be tied together due to the check when removing a graph edge. If you do the replacement first, that check fails as the output node now has an edge to a replacement. So you need to get the info about the edges, remove them, and then create the replacement ones. You could separate it into two methods but now you have to expose the GraphEdge type, and the user of the interface needs to make 3 calls to do this operation. 

Happy to rename, although it's not connecting the incoming node with the outgoing nodes - it's connecting an arbitrary replacement node with the outgoing nodes.

Reply:
Closing outdated comments